fuzzers/config_file_fuzzer.c

```diff
@@ -0,0 +1,41 @@
+#include <git2.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <limits.h>
+
+int foreach_cb(const git_config_entry *entry, void *payload) {
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    static int fd = -1;
+    static char path[] = "/tmp/git.XXXXXX";
+    if (fd < 0) {
+        git_libgit2_init();
+        fd = mkstemp(path);
+        if (fd < 0) {
+            abort();
+        }
+    }
+    if (ftruncate(fd, 0) !=0 ) {
+        abort();
+    }
+    if (lseek(fd, 0, SEEK_SET) != 0) {
+        abort();
+    }
+    if (write(fd, data, size) != size) {
+        abort();
+    }
+
+    git_config *cfg;
+    int err = git_config_open_ondisk(&cfg, path);
```

Comment:
Our project uses the C89 standard, where declarations after statements aren't allowed. You'll have to move all variable declarations to the head of this function

Reply:
This is not your fault, but it really sucks that libgit2 isn't yet able to open configs from memory. I wanted to refactor that part for a long time already to make it possible, but until now I didn't yet have enough time to do this. I bet that writing a file for each iteration will slow the fuzzer down by quite a bit.